libs/backend/src/scan/cast_vote_records/cast_vote_record_exporter.ts

```diff
@@ -0,0 +1,415 @@
+  | { type: 'invalid-sheet-found'; message: string }
+  | ExportDataError;
+
+/**
+ * The subset of scanner store methods relevant to exporting cast vote records
+ */
+export interface ScannerStore {
+  clearCastVoteRecordHashes(): void;
+  getBatches(): BatchInfo[];
+  getCastVoteRecordRootHash(): string;
+  getDefiniteMarkThreshold(): number;
+  getElectionDefinition(): ElectionDefinition | undefined;
+  getExportDirectoryName(): string | undefined;
+  getMode(): 'live' | 'test';
+  getPollsState(): PollsState | undefined;
+  updateCastVoteRecordHashes(
+    castVoteRecordId: string,
+    castVoteRecordHash: string
+  ): void;
+  setExportDirectoryName(exportDirectoryName: string): void;
+}
+
+/**
+ * State that can be retrieved via the ScannerStore interface and that is unchanged by exporting
+ * cast vote records
+ */
+interface ScannerStateUnchangedByExport {
+  batches: BatchInfo[];
+  definiteMarkThreshold: number;
+  electionDefinition: ElectionDefinition;
+  inTestMode: boolean;
+  pollsState?: PollsState;
+}
+
+interface File {
+  fileName: string;
+  fileContents: string | Buffer;
+}
+
+/**
+ * Metadata stored in the top-level metadata file for a cast vote record export
+ */
+interface CastVoteRecordExportMetadata {
+  arePollsClosed?: boolean;
+  /** Global data relevant to all cast vote records in an export, e.g. election info */
+  castVoteRecordReportMetadata: CVR.CastVoteRecordReport;
+  /** A hash of all cast vote record files in an export */
+  castVoteRecordRootHash: string;
+}
+
+/**
+ * The API for a cast vote record exporter that a scanner can use to export cast vote records.
+ * Supports both continuous and batch export use cases.
+ */
+export interface CastVoteRecordExporterApi {
+  exportCastVoteRecordsToUsbDrive(
+    resultSheets: ResultSheet[] | Generator<ResultSheet>
+  ): Promise<Result<void, ExportCastVoteRecordToUsbDriveError>>;
+}
+
+/**
+ * The implementation of the cast vote record exporter API
+ */
+export class CastVoteRecordExporter implements CastVoteRecordExporterApi {
+  private readonly artifactAuthenticator: ArtifactAuthenticatorApi;
+  private readonly exporter: Exporter;
+  private readonly scannerStore: ScannerStore;
+  private readonly usbDrive: UsbDrive;
+
+  constructor(input: {
+    artifactAuthenticator: ArtifactAuthenticatorApi;
+    scannerStore: ScannerStore;
+    usbDrive: UsbDrive;
+  }) {
+    this.artifactAuthenticator = input.artifactAuthenticator;
+    this.scannerStore = input.scannerStore;
+    this.usbDrive = input.usbDrive;
+
+    this.exporter = new Exporter({
+      allowedExportPatterns: SCAN_ALLOWED_EXPORT_PATTERNS,
+      getUsbDrives: async () => {
+        const drive = await this.usbDrive.status();
+        return drive.status === 'mounted' ? [drive] : [];
+      },
+    });
+  }
+
+  async exportCastVoteRecordsToUsbDrive(
+    resultSheets: ResultSheet[] | Generator<ResultSheet>
+  ): Promise<Result<void, ExportCastVoteRecordToUsbDriveError>> {
```

Comment:
This function mirrors the existing [exportCastVoteRecordReportToUsbDrive](https://github.com/votingworks/vxsuite/blob/325446e15f477c4587a644b5a238df6902e5d1d6/libs/backend/src/scan/cast_vote_records/export.ts#L481-L484) but is designed for continuous export. The structure of the export directory now looks something like this:

```
cast-vote-records
└── test-ballot_general-election_5fb429f84c
    ├── TEST__machine_000__2023-08-17_10-47-02
    │   ├── 0c976e40-9ac5-468b-b167-fe7fa37fab38
    │   │   ├── 0c976e40-9ac5-468b-b167-fe7fa37fab38-back.jpg
    │   │   ├── 0c976e40-9ac5-468b-b167-fe7fa37fab38-back.layout.json
    │   │   ├── 0c976e40-9ac5-468b-b167-fe7fa37fab38-front.jpg
    │   │   ├── 0c976e40-9ac5-468b-b167-fe7fa37fab38-front.layout.json
    │   │   └── cast-vote-record-report.json
    │   ├── 2d902f55-4a02-4a9a-922b-d01f8213010e
    │   │   ├── 2d902f55-4a02-4a9a-922b-d01f8213010e-back.jpg
    │   │   ├── 2d902f55-4a02-4a9a-922b-d01f8213010e-back.layout.json
    │   │   ├── 2d902f55-4a02-4a9a-922b-d01f8213010e-front.jpg
    │   │   ├── 2d902f55-4a02-4a9a-922b-d01f8213010e-front.layout.json
    │   │   └── cast-vote-record-report.json
    │   ├── cf5a6f3b-a7c1-4f1b-8af5-b969dbe62f41
    │   │   ├── cast-vote-record-report.json
    │   │   ├── cf5a6f3b-a7c1-4f1b-8af5-b969dbe62f41-back.jpg
    │   │   ├── cf5a6f3b-a7c1-4f1b-8af5-b969dbe62f41-back.layout.json
    │   │   ├── cf5a6f3b-a7c1-4f1b-8af5-b969dbe62f41-front.jpg
    │   │   └── cf5a6f3b-a7c1-4f1b-8af5-b969dbe62f41-front.layout.json
    │   └── metadata.json
    └── TEST__machine_000__2023-08-17_10-47-02.vxsig
```

In VxScan, the common case will be to pass a result sheets array of length 1 right after scanning a ballot (as you'll see in a subsequent commit). But we still want to support passing a generator capable of iterating over all result sheets for full exports, e.g. on VxCentralScan or on VxScan when using the EM button to export all CVRs.